src/main/java/br/unicamp/meca/models/ActionSequencePlan.java

```diff
@@ -44,22 +44,36 @@ public ActionSequencePlan(ActionStep[] actionIdSequence) {
 	}
 	
 	/**
-	 * Returns the id of the current ActionFromPlanningCodelet to be undertaken in the plan.
+	 * Returns the ActionStep of the current ActionFromPlanningCodelet to be undertaken in the plan.
 	 * 
 	 * @return currentActionId
 	 */
 	public ActionStep getCurrentActionStep() {
 		return actionIdSequence[currentActionIdIndex];
 	}
+        
+        /**
+	 * Returns the ActionStep of the last ActionFromPlanningCodelet to be undertaken in the plan, or null if it is the first step.
+	 * 
+	 * @return currentActionId
+	 */
+	public ActionStep getLastActionStep() {
+            if (currentActionIdIndex == 0) return(actionIdSequence[actionIdSequence.length-1]);
+            else return actionIdSequence[currentActionIdIndex-1];
+	}
 
 	/**
+         * get the array of ActionSteps which are related to the ActionSequencePlan
+         * 
 	 * @return the actionIdSequence
 	 */
 	public ActionStep[] getActionStepSequence() {
 		return actionIdSequence;
 	}
 
 	/**
+         * Forces a new actionStepSequence
+         * 
 	 * @param actionIdSequence the actionIdSequence to set
 	 */
 	public void setActionStepSequence(ActionStep[] actionIdSequence) {
```

Comment:
The parameter name is still refering to the old `id`, but now it is an array. Javadoc needs updating as well.

Reply:
Changed